lib/sentry/transport.ex

```diff
@@ -86,6 +86,22 @@ defmodule Sentry.Transport do
     kind, data -> {:error, {kind, data, __STACKTRACE__}}
   end
 
+  defp post_validation(client, endpoint, headers, body) do
```

Comment:
We should also validate that the status is an integer, the body is a string, and headers are a list.

```elixir
case client.post(...) do
  {:ok, status, resp_headers, resp_body} when is_integer(status) and status in 200..599 and is_list(resp_headers) and is_binary(resp_body) ->
```